### ISSUE TITLE
"prepublish" is deprecated

### ISSUE THREAD

**Issue by stevage** [perliedman/leaflet-control-geocoder#175]
Package.json contains:

`"prepublish": "sh ./scripts/build.sh",`

Which is deprecated:

```
npm WARN prepublish-on-install As of npm@5, `prepublish` scripts are deprecated.
npm WARN prepublish-on-install Use `prepare` for build steps and `prepublishOnly` for upload-only.
npm WARN prepublish-on-install See the deprecation note in `npm help scripts` for more information.
```

This script fails to run on my machine (OSX):

```
> leaflet-control-geocoder@1.5.6 prepublish /Users/steveb/dev/leaflet-control-geocoder
> sh ./scripts/build.sh

Error: Cannot find module 'leaflet' from '/Users/steveb/dev/leaflet-control-geocoder/test'
    at /Users/steveb/dev/leaflet-control-geocoder/node_modules/browser-resolve/node_modules/resolve/lib/async.js:46:17
    at process (/Users/steveb/dev/leaflet-control-geocoder/node_modules/browser-resolve/node_modules/resolve/lib/async.js:173:43)
    at ondir (/Users/steveb/dev/leaflet-control-geocoder/node_modules/browser-resolve/node_modules/resolve/lib/async.js:188:17)
    at load (/Users/steveb/dev/leaflet-control-geocoder/node_modules/browser-resolve/node_modules/resolve/lib/async.js:69:43)
    at onex (/Users/steveb/dev/leaflet-control-geocoder/node_modules/browser-resolve/node_modules/resolve/lib/async.js:92:31)
    at /Users/steveb/dev/leaflet-control-geocoder/node_modules/browser-resolve/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:123:15)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
```

So it would be good to know whether this is actually required for an end user, or is a real "pre-publish" step.

**Comment by perliedman** [comment]
Thanks for pointing this out. The script is indeed needed on publish as well as on local install, since it produces the `dist` files, which are the modules `main`.

I think you're seeing the error since you're not running the script from the npm context with paths set up correct etc.; does it work for you when you do `npm install`? If not, please open a new issue for this.

**Comment by stevage** [comment]
It failed when I ran `"npm install"` from within the source directory.

**Comment by perliedman** [comment]
Ok, I found a stupid leftover in the build script that I'm pretty certain caused this.

I have removed it and released 1.5.7 with this and a few other fixes.
